### DISCUSSION TITLE
Regex syntax for window rules

### DISCUSSION THREAD

**Discussion by raffaelemancuso** [hyprwm/Hyprland#1526] [Q&A]
What is the regex syntax for window rules? POSIX? Perl?

Are lookahead and lookbehind assertions supposed to work?

**Comment by vaxerski** [comment]
it's whichever C++ uses. Seems to be `ECMA-262`, which does not have lookbehind.

**Comment by raffaelemancuso** [comment]
I have to match the following window:

```
Window ba563b70 -> PARTENZA:
	at: 968,45
	size: 939,1022
	workspace: 3 (3)
	floating: 0
	monitor: 0
	class: betterbird
	title: PARTENZA
	pid: 418184
	xwayland: 0
	pinned: 0
	fullscreen: 0
	fullscreenmode: 0
	fakefullscreen: 0
	grouped: 0
	swallowing: 0
```

Without matching the following one:
```
Window ba55aad0 -> Calendar - Betterbird:
	at: 13,45
	size: 939,1022
	workspace: 3 (3)
	floating: 0
	monitor: 0
	class: betterbird
	title: Calendar - Betterbird
	pid: 418184
	xwayland: 0
	pinned: 0
	fullscreen: 0
	fullscreenmode: 0
	fakefullscreen: 0
	grouped: 0
	swallowing: 0
```

How can I do it?

**Reply by vaxerski** [reply]
`windowrulev2=somerule,class:^(betterbird)$,title:^(PARTENZA)$`

**Reply by raffaelemancuso** [reply]
No, "Partenza" can change.

Basically I use betterbird to manage my calendar. I want the betterbird app to be maximized, but when I click on an event, I want the event to show in a popup.

"Partenza" is just the name of that particular event. It can change.

The only stable thing seems to be that the "app" window, that is, the non-popup windows, have "Betterbird" in their title at the end.

**Reply by vaxerski** [reply]
`^(?!.*\ \-\ Betterbird).*$`

**Reply by raffaelemancuso** [reply]
It is working indeed. So it doesn't have lookbehind, but it does have lookahead and in particular negative lookahead assertion?

I don't understand

**Reply by raffaelemancuso** [reply]
It looks [like so](https://262.ecma-international.org/5.1/#sec-15.10.2.8). It has positive lookahead and negative lookahead, but no lookbehind

**Comment by GlennisOlienocu** [comment]
it looks like they're making some progress!
